pkg/helper/airgap.go

```diff
@@ -0,0 +1,174 @@
+package helper
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/pkg/logger"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+var log = logger.AddLogger()
+
+func SetupBastion(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	log.Infof("Downloading %v artifacts...", cluster.Config.Product)
+	_, err := getArtifacts(cluster, flags)
+	Expect(err).To(BeNil())
+
+	setupPrivateRegistry(cluster, flags)
+	dockerActions(cluster, flags)
+
+	pwd, err := shared.RunCommandOnNode("pwd", cluster.GeneralConfig.BastionIP)
+	Expect(err).To(BeNil())
+
+	regMap := map[string]string{
+		"$PRIVATE_REG": cluster.GeneralConfig.BastionDNS,
+		"$USERNAME":    flags.AirgapFlag.RegistryUsername,
+		"$PASSWORD":    flags.AirgapFlag.RegistryPassword,
+		"$HOMEDIR":     pwd,
+	}
+
+	updateRegistryFile(cluster, regMap)
+}
+
+func setupPrivateRegistry(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	log.Info("Setting up private registry...")
+	cmd := fmt.Sprintf(
+		"sudo chmod +x private_registry.sh && "+
+			`sudo ./private_registry.sh "%v" "%v" "%v"`,
+		flags.AirgapFlag.RegistryUsername, flags.AirgapFlag.RegistryPassword,
+		cluster.GeneralConfig.BastionDNS)
+	_, err := shared.RunCommandOnNode(cmd, cluster.GeneralConfig.BastionIP)
+	Expect(err).To(BeNil())
+}
+
```

Comment:
on this file, any of those functions do you think would be nice to be more abstract/generic so we could potentially use it somewhere else ?

Reply:
Yes that is the goal as this is only private registry and some func here can be reused for other airgap install methods